src/app/components/admin/analysis-jobs/list/list.component.ts

```diff
@@ -0,0 +1,84 @@
+import { Component, OnInit } from "@angular/core";
+import { AnalysisJobsService } from "@baw-api/analysis/analysis-jobs.service";
+import {
+  adminAnalysisJobsMenuItem,
+  adminDashboardMenuItem,
+} from "@components/admin/admin.menus";
+import { adminMenuItemActions } from "@components/admin/dashboard/dashboard.component";
+import { PagedTableTemplate } from "@helpers/tableTemplate/pagedTableTemplate";
+import { Id, Param } from "@interfaces/apiInterfaces";
+import { AnalysisJob } from "@models/AnalysisJob";
+import { List } from "immutable";
+import {
+  adminAnalysisJobMenuItem,
+  adminAnalysisJobsCategory,
+} from "../analysis-jobs.menus";
+
+@Component({
+  selector: "baw-admin-analysis-jobs",
+  templateUrl: "./list.component.html",
+})
+class AdminAnalysisJobsComponent
+  extends PagedTableTemplate<TableRow, AnalysisJob>
+  implements OnInit {
+  public columns = [
+    { name: "Id" },
+    { name: "Name" },
+    { name: "Script" },
+    { name: "Creator" },
+    { name: "Started" },
+    { name: "Status" },
+    { name: "Status Updated" },
+    { name: "Actions" },
+  ];
+  public sortKeys = {
+    id: "id",
+    name: "name",
+    started: "startedAt",
+    status: "overallStatus",
+    statusUpdated: "overallStatusModifiedAt",
+  };
+
+  constructor(api: AnalysisJobsService) {
+    super(api, (analysisJobs) =>
+      analysisJobs.map((analysisJob) => ({
+        id: analysisJob.id,
+        name: analysisJob.name,
+        script: analysisJob.scriptId,
+        creator: analysisJob.creatorId,
+        started: analysisJob.startedAt.toRelative(),
+        status: analysisJob.overallStatus,
+        statusUpdated: analysisJob.overallStatusModifiedAt.toRelative(),
+        model: analysisJob,
+      }))
+    );
+  }
```

Comment:
This paged table template seems very noisy.

- mapping of models is redundant (mostly)
- column names are verbose
- informal view models, and duplicated objects

It seems to me we should pass:

- models
- table options
    - columns? : ColumnOptions[] // optional: shows some columns be default, like `id` and `name`
        - name: string
        -  { name: string, displayName?: string, transform?: (x: T) => Y }

Reply:
`columns` object is annoying defined that way by the datatable library

Reply:
I could rewrite this to clean it up and reduce the amount of code, however all of the admin pages are done, and changing it would just be spending time cleaning up pages that are:

a) working
b) not important (for the most part, some even appear unused)
c) unlikely to change much in the future
d) changes would require re-implementing/testing all of the admin list pages

I would suggest instead creating an issue with my ideas on how to re-implement this code

Reply:
Added to #457